mne/io/eeglab/_eeglab.py

```diff
@@ -71,13 +71,90 @@ def _check_for_scipy_mat_struct(data):  # taken from pymatreader.utils
     return data
 
 
-def _readmat(fname, uint16_codec=None):
+def _readmat(fname, uint16_codec=None, preload=False):
     try:
         read_mat = _import_pymatreader_funcs("EEGLAB I/O")
     except RuntimeError:  # pymatreader not installed
-        eeg = loadmat(
-            fname, squeeze_me=True, mat_dtype=False, uint16_codec=uint16_codec
-        )
+        if preload:
+            eeg = loadmat(
+                fname, squeeze_me=True, mat_dtype=False, uint16_codec=uint16_codec
+            )
+        else:
+            info_fields = [
+                "setname",
```

Comment:
Is there a list available (like an EEGLAB API documentation) that describes or at least simply but exhaustively enumerates  these fields? If yes, it would be very valuable to link to that in a comment here, or in the docstr of the EEGLAB reading function. If not, I expect this bit of code to be relatively volatile, as we would always need an EEGLAB expert to help us fixing this function.